rts/Lua/LuaMatrixImpl.cpp

```diff
@@ -0,0 +1,323 @@
+#include "LuaMatrixImpl.h"
+
+#include <algorithm>
+
+#include "LuaUtils.h"
+
+#include "Sim/Misc/LosHandler.h"
+#include "Sim/Objects/SolidObject.h"
+#include "Sim/Projectiles/Projectile.h"
+#include "Sim/Projectiles/ProjectileHandler.h"
+#include "Sim/Features/Feature.h"
+#include "Sim/Features/FeatureHandler.h"
+#include "Sim/Units/Unit.h"
+#include "Sim/Units/UnitHandler.h"
+#include "Rendering/Models/3DModel.h"
+#include "Rendering/ShadowHandler.h"
+#include "Rendering/GlobalRendering.h"
+#include "Game/Camera.h"
+#include "Game/CameraHandler.h"
+
+CMatrix44f LuaMatrixImpl::screenViewMatrix = CMatrix44f{};
+CMatrix44f LuaMatrixImpl::screenProjMatrix = CMatrix44f{};
+
+void LuaMatrixImpl::Zero()
+{
+	std::fill(&mat[0], &mat[0] + 16, 0.0f);
+}
+
+void LuaMatrixImpl::SetMatrixElements(
+	const float  m0, const float  m1, const float  m2, const float  m3,
+	const float  m4, const float  m5, const float  m6, const float  m7,
+	const float  m8, const float  m9, const float m10, const float m11,
+	const float m12, const float m13, const float m14, const float m15)
+{
+	const std::array<float, 16> arr {m0, m1, m2, m3, m4, m5, m6, m7, m8, m9, m10, m11, m12, m13, m14, m15};
+	std::copy(arr.cbegin(), arr.cend(), &mat.m[0]);
```

Comment:
I've a feeling CMatrix44f could use a constructor for this.
Something akin to 
```
    CMatrix44f(const float  m0, const float  m1, const float  m2, const float  m3,
	const float  m4, const float  m5, const float  m6, const float  m7,
	const float  m8, const float  m9, const float m10, const float m11,
	const float m12, const float m13, const float m14, const float m15) : m{m0, m1, m2, m3, m4, m5, m6, m7, m8, m9, m10, m11, m12, m13, m14, m15}
    {}
```
Then you can just
```
mat = CMatrix44f { m0, m1, m2, m3, m4, m5, m6, m7, m8, m9, m10, m11, m12, m13, m14, m15 };
```